containers/buildah
# buildah: actually use containers.conf settings

#### What type of PR is this?




 /kind bug
 /kind cleanup

#### What this PR does / why we need it:

Buildah ignores the container.conf settings. Commit 05d5d22dc6b7 tried
to fix this but I cannot see how this would have worked, there were also
no test added related to containers.conf.

The code below were we read the default just overwrites everything we
already set so the config settings are ignored.

Also there are many weird places were settings are just overwritten for
no particular reason. The namespaces code path looks like mess to me.
Namespaces are added/changed/removed in many different places and there
is no explanation why. Maybe I am just not familar enough with this code
base but I cannot really understand this. Hopefully a maintainer can
take a closer look to see if my changes are indeed correct.

#### How to verify it

add
```
[containers]
netns = "host"
```
to container.conf add run buildah build and check the namespace.

#### Which issue(s) this PR fixes:

Fixes containers/podman#13294

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?



```release-note

```
